week-1/L-functions-nested/exercise2.js

```diff
@@ -0,0 +1,25 @@
+/*
+
+*I feel this code all looks very hacky.
+
+*/
+
+var mentor1 = "Daniel";
+var mentor2 = "Irina";
+var mentor3 = "Mimi";
+var mentor4 = "Rob";
+var mentor5 = "Yohannes";
+
+var array = [mentor1, mentor2, mentor3, mentor4, mentor5]
+var upperCase = array.map(function(arrayobject) {
+  return arrayobject.toUpperCase()
+})
+
+function greeting() {
+  message = 'HELLO' + ' ' + upperCase[i]
+  return message
+}
+
+  for (var i = 0; i < upperCase.length; i++) {
+    console.log(greeting(upperCase[i]))
+  }
```

Comment:
Great piece of work. I have just discovered an alternative way, I didn't know about array.map.  Why did you feel the code looks hacky?

Reply:
Hey Tayo! 

Thanks for your nice comments. I guess I thought it was hacky of how I came up with the solution. I decided I wanted to do the console log with a loop statement and because of that shaped the whole code around it. This led me having to do some workarounds just to get the code to work how I wanted it to. An example of this was using the .map method as you can't turn a whole array into uppercase without it.